env_spec.py

```diff
@@ -177,32 +186,36 @@ def parse(env_spec_text):
 
                 if choices_match:
                     if default_value not in choices or default_value == "":
-                        raise EnvSpecSyntaxError("SYNTAX ERROR: Invalid default value.")
+                        raise EnvSpecSyntaxError(
+                            "SYNTAX ERROR: Invalid default value.", line_number
+                        )
 
                 env_spec_entry["default_value"] = default_value
 
             if not choices_match:
                 if default_value_match:
                     env_spec_type = default_value_match.groups()[0]
                 else:
-                    env_spec_type = line
+                    env_spec_type = env_spec_line
 
                 env_spec_type = env_spec_type.strip()
 
                 if env_spec_type not in valid_types_list:
-                    raise EnvSpecSyntaxError("SYNTAX ERROR: Invalid type.")
+                    raise EnvSpecSyntaxError("SYNTAX ERROR: Invalid type.", line_number)
```

Comment:
Can you change the error message to this please?

```
Invalid variable type; it should be one of "bla", "blah", "blah".
```

**P.S.:** @krantou do not hardcode the valid types in the error message above, but construct the substring by joining the elements of `valid_types_list`.